docs/ruleset_functions/compare_swh_dist_systems_and_components.md

```diff
@@ -0,0 +1,54 @@
+  - In addition to the equipment connected to the distribution system, there's also equipment that's part of the service water heating distribution system that are not direct child objects of the distribution system.  We need to check these objects.
+  - first, check ServiceWaterHeatingEquipment - when we execute compare_context_pair, this will also check any child objects that exist (SolarThermal and SWH validation point).  Start by checking if there are the same number of objects in models 1 and 2.  We need to do the length check here because it's not checked implicitly as part of compare_context_pair.  For example, if there are more pieces of equipment in the model 1 than model 2, comparing each item found in model 2 could return a false positive: `if len(RMD1_swh_equipment_dict["SWHHeatingEq"]) == len(RMD2_swh_equipment_dict["SWHHeatingEq"]):`
+    - look at each SWHEquipment in the model 1: `for swh_eq_id in RMD1_swh_equipment_dict["SWHHeatingEq"]:`
+      - get the SWH equipment for models 1 and 2: `swh_eq_1 = get_component_by_id(RMD1, swh_eq_id); swh_eq_2 = get_component_by_id(RMD2, swh_eq_id)`
+      - compare the two SWH equipment using compare_context_pair.  By passing in the same errors list, any new errors get appended to the list.  If the comparison fails, we add an error to the errors list indicating the id of the swh equipment that failed comparison: `if !compare_context_pair(swh_eq_1, swh_eq_2, $, extra_schema_for_SWH_comparison.json, true, compare_context_str, errors): errors << "Comparison of SWH Equipment: " + swh_eq_id + " failed."`
+  - otherwise there is an unequal number of equipment entries.  Add our own error to the errors list: `else: errors << "Unequal numbers of SWH Equipment between the two models for " + swh_distribution_id`
+  - next, check Pumps - this will also recursively check PumpOutputValidationPointPumpOutputValidationPoint: `if len(RMD1_swh_equipment_dict["Pumps"]) == len(RMD2_swh_equipment_dict["Pumps"]):`
+    - look at each SWHEquipment in the proposed model: `for pump_id in RMD1_swh_equipment_dict["Pumps"]:`
+      - get the pumps for models 1 and 2: `pump_1 = get_component_by_id(RMD1, pump_id); pump_2 = get_component_by_id(RMD2, pump_id)`
+      - compare the two pumps using compare_context_pair.  If the comparison fails, we add an error to the errors list indicating the id of the swh pump that failed comparison: `if !compare_context_pair(pump_1, pump_2, $, extra_schema_for_SWH_comparison.json, true, compare_context_str, error_str): errors << "Comparison of SWH Pump: " + pump_id + " failed."`
```

Comment:
I'm not sure why this is checking `if !compare_context_pair`. If the function returns a list then the list will behave as a truthy value

Reply:
old logic - I've removed if !